encrypted-media/scripts/clearkey-update-non-ascii-input.js

```diff
@@ -0,0 +1,42 @@
+//this test is only applicable to clearkey
+function runTest(config, qualifier) {
+    var testname = testnamePrefix(qualifier, config.keysystem) + ' test handling of non-ASCII responses for update()';
+
+    var configuration = getSimpleConfigurationForContent(config.content);
+
+    if (config.initDataType && config.initData) configuration.initDataTypes = [config.initDataType];
+
+    async_test(function (test) {
+        var initDataType;
+        var initData;
+        var mediaKeySession;
+
+        //clearkey JSON Web Key needs to be Ascii encoded
+        function processMessage(event) {
+                // |jwkSet| contains a  non-ASCII character \uDC00.
+                var jwkSet = '{"keys":[{'
+                    +     '"kty":"oct",'
+                    +     '"k":"MDEyMzQ1Njc4OTAxMjM0NQ",'
+                    +     '"kid":"MDEyMzQ1Njc4O\uDC00TAxMjM0NQ"'
+                    + '}]}';
+
+
+                event.target.update(stringToUint8Array(jwkSet)).then(function () {
+                    forceTestFailureFromPromise(test, 'Error: update() succeeded');
```

Comment:
nit: The description here is a bit confusing, since it is not clear why success of update() is an error. Could you modify it a bit to smth like: "Error: update() should fail because the processed message has non-ASCII character."